doc/cache_layout.md

```diff
@@ -0,0 +1,243 @@
+# The Pub cache
+
+The Pub cache is where pub stores downloaded packages and globally activated
+packages.
+
+The information in this document is informational, and can be used for
+understanding the cache, but we strongly encourage all manipulation of the cache
+happens though the `dart pub`/`flutter pub` commands to avoid relying on
+accidental properties of the cache that might be broken in the future.
+
+See [system_cache](../lib/src/system_cache.dart) for implementation of top-level
+cache conventions.
+
+## Location
+
+For the Dart SDK the default pub cache location is `$HOME/.pub_cache` on Linux
+and Mac OS, and `%LOCALAPPDATA%/Pub/Cache` on Windows.
+
+For the Flutter SDK after 3.8 uses the same folder. Before it would use the same
+folder if it was found, and otherwise  use `$FLUTTER_ROOT/.pub_cache`.
+
+The environment variable `PUB_CACHE` can be used to override the location of the
+pub cache.
+
+In this document we refer to this as `.pub_cache`.
+
+## Layout
+
+The layout of the pub cache has evolved over time, and where possible we strive
+for backwards and forwards compatibility where possible, such that a new and an
+old sdk can share the same cache.
+
+Here are the top-level folders you can find in a Pub cache.
+
+```plaintext
+.pub-cache/
+├── global_packages # Globally activated packages
+├── bin # Executables compiled from globally activated packages.
+├── git # Cloned git packages
+├── hosted # Hosted package downloads
+├── hosted-hashes # Hashes of hosted packages
+├── log # Logs after crashes and --verbose
+├── README.md # Short description of the folder
+└── _temp # Package downloads are extracted here, and moved atomically.
+```
+
+Before Dart 2.15 pub would also store credentials in the pub cache. They are now
+stored in a platform specific config dir:
+
+* On Linux $XDG_CONFIG_HOME/dart/pub-credentials.json if $XDG_CONFIG_HOME is
+  defined, otherwise $HOME/.config/dart/pub-credentials.json
+* On Mac OS: $HOME/Library/Application Support/dart/pub-credentials.json
+* On Windows: %APPDATA%/dart/pub-credentials.json
```

Comment:
```suggestion
* On Linux `$XDG_CONFIG_HOME/dart/pub-credentials.json` if `$XDG_CONFIG_HOME` is
  defined, otherwise `$HOME/.config/dart/pub-credentials.json`
* On Mac OS: `$HOME/Library/Application Support/dart/pub-credentials.json`
* On Windows: `%APPDATA%/dart/pub-credentials.json`
```